### PR TITLE
Issues 11.135,136,141

### PR DESCRIPTION
**Pull request by sunilmeddr**

### CODE REVIEW

**Comment by igarashitm** [comment]
Hi @sunilmeddr ,

Can you describe what has been done specifically in the commit log so we can trace later? Also you can put `Fixes: #11` to close the issue automatically when it's merged. If you still want to keep some of them open, then just mention with hash. like this (just an example, I haven't checked if it resolves #141 or not):
```
fix: Migrate XmlPathCoordinate and PathUtil into AtlasPath, added XmlPath with extending AtlasPath
fix: something else you've done...

Fixes: #11
Fixes: #135
Fixes: #136
Also Addresses: #141
```

**Comment by mattrpav** [comment]
Changes LGTM

+1 update the commit comments and I think its good to merge

@igarashitm thoughts?

**Comment by sunilmeddr** [comment]
Commit message is updated

File: runtime/core/src/main/java/io/atlasmap/core/PathUtil.java [hunk 1]
```diff
@@ -1,483 +0,0 @@
...
-
-    public static boolean isMapSegment(String segment) {
-        return isCollectionSegment(segment) && segment.contains(PATH_MAP_START);
-    }
-
-    public static String findNextSegment(String segment, String path) {
-        PathUtil pathUtil = new PathUtil(path);
-        String parentSegment = null;
-        for (String currentSegment : pathUtil.segments) {
-            if (parentSegment == null || !segment.equals(parentSegment)) {
```
Comment (igarashitm) [inline runtime/core/src/main/java/io/atlasmap/core/PathUtil.java]: Yeah I think Codacy is right here, this loop looks super weird. @sunilmeddr I know it was just copied, but can you turn over the condition and just return in 'if'?
https://www.codacy.com/app/atlasmapio/atlasmap/pullRequest?prid=1131462

File: runtime/core/src/main/java/io/atlasmap/core/AtlasPath.java [hunk 2]
```diff
@@ -230,44 +317,170 @@ public String getCollectionSegment() {
...
+
+    public static boolean isMapSegment(String segment) {
+        return isCollectionSegment(segment) && segment.contains(PATH_MAP_START);
+    }
+
+    public static String findNextSegment(String segment, String path) {
+        AtlasPath pathUtil = new AtlasPath(path);
+        String parentSegment = null;
+        for (String currentSegment : pathUtil.segments) {
+            if (parentSegment == null || !segment.equals(parentSegment)) {
```
Comment (igarashitm) [inline runtime/core/src/main/java/io/atlasmap/core/AtlasPath.java]: Yeah I think Codacy is right here, this loop looks super weird. @sunilmeddr I know it was just copied, but can you turn over the condition and just return in 'if'?
https://www.codacy.com/app/atlasmapio/atlasmap/pullRequest?prid=1131462

**Comment by igarashitm** [comment]
Ah you put the commit log message onto follow-up commit - you may want to squash into single commit then? GitHub Blame is a powerful tool to trace the code later and it relies on appropriate commit message.

**Comment by sunilmeddr** [comment]
Method findNextsegment has no references,So I deleted the method.

**Comment by igarashitm** [comment]
I meant the commit `Updated for issues 11.135,136,141` contains major part of your work for this, which should have an appropriate commit log. I would rather squash into a single commit though.

**Comment by igarashitm** [comment]
Wait, please don't include other commit which has no relation with the topic, path handling. `StringConverter should be more forgiving for conversion to Boolean` is totally different topic, so it must be a separate pull request.

**Pull request merged**